Classes/Controller/AbstractController.php

```diff
@@ -350,16 +350,77 @@ protected function buildSimplePagination(PaginationInterface $pagination, Pagina
     {
         $firstPage = $pagination->getFirstPageNumber();
         $lastPage = $pagination->getLastPageNumber();
+        $currentPageNumber = $paginator->getCurrentPageNumber();
+
+        $pages = [];
+
+        $lastStartRecordNumberGrid = 0; // due to validity outside the loop
+        foreach (range($firstPage, $lastPage) as $i) {
+            // detect which pagination is active: ListView or GridView
+            if (get_class($pagination) == 'TYPO3\CMS\Core\Pagination\SimplePagination') {  // ListView
+                $lastStartRecordNumberGrid = $i; // save last $startRecordNumber for LastPage button
+
+                $pages[$i] = [
+                    'label' => $i,
+                    'startRecordNumber' => $i
+                ];
+            } else { // GridView
+                // to calculate the values for generation the links for the pagination pages
+                $itemsPerPage = $paginator->getPublicItemsPerPage();
```

Comment:
This should hopefully fix the PHPStan error:
```suggestion
                /** @var Kitodo\Dlf\Pagination\PageGridPaginator $paginator */
                $itemsPerPage = $paginator->getPublicItemsPerPage();
```

Reply:
I have added your suggestion, but unfortunately it does not quite solve the problem. There is still an error message:

`PHPDoc tag @var for variable $paginator contains unknown class Kitodo\Dlf\Controller\Kitodo\Dlf\Pagination\PageGridPaginator.`

Reply:
My fault: The namespace should be absolute, i. e. with a prefixed backslash. I fixed my proposal above.